XenAdmin/Controls/Ballooning/HostMemoryControls.cs

```diff
@@ -127,8 +127,11 @@ void vm_metrics_PropertyChanged(object sender, PropertyChangedEventArgs e)
 
         private void valueControlDomain_LinkClicked(object sender, LinkLabelLinkClickedEventArgs e)
         {
-            using (var dlg = new ControlDomainMemoryDialog(host))
-                dlg.ShowDialog(Program.MainWindow);
+            if (host != null)
+            {
+                using (var dlg = new ControlDomainMemoryDialog(host))
+                    dlg.ShowDialog(Program.MainWindow);
+            }

```

Comment:
Nit: You could argue that an early exit would be even clearer here i.e.:
```csharp
if (host == null)
    return;

using (var dlg = new ControlDomainMemoryDialog(host))
    dlg.ShowDialog(Program.MainWindow);
```

Not a bit deal either way though.

Reply:
Might as well, since we have so many commits in this one already.